### DISCUSSION TITLE
Library display seems to be different between V1.8.2 and V1.9.2

### DISCUSSION THREAD

**Discussion by BobManMasher** [gerbera/gerbera#2290] [Q&A] [answered]
Now I have V1.9.2 working on an identical copy of the media files, I have noticed a difference in the way my album information is displayed in Bubble Media player.  When I request a list of all albums, I get a list which includes the album artist and so bubble can sort on that field from the V1.8.2 but this doesn't come through on 1.9.2  Both servers have the option set to display album artist.

I also seem to have lost all my album covers unless htey are embeded into the mp3 audio files.  The album art is usually called something like cover.jpg and there is an image in virtually every album folder. Not sure at what point they disappeared but they are no more now.

I dont have scripting enabled in either version and the script files look identical from a cursory glance.  I cant see a setting in config.xml either so any pointers would be appreciated.

Thanks - Bob

**Comment by KarlStraussberger** [comment, marked as answer]
Yes, there was a change in handling external resources.

You need to set it up https://docs.gerbera.io/en/stable/config-import.html#resources now. 1.8.2 was looking for hard coded file names.

**Reply by BobManMasher** [reply]
Thanks Karl - I have to admit that the page makes no sense to me and when I added the example to my config.xml my gerbera wouldn't start. I think it may be a cut and paste  problem as I then found another example on a different part of the forum (which looked identical) and it now scans and I can see the album art when I look at the database 'PC Directory' but not when I look at the database 'audio -> albums' . The fanart is also not being downloaded into vlc  or bubble.  I get the same result on 1.8.2 and 1.9.2 so I'm further forward but not quite there yet.  I have created a section for <container> and <resource> but I don't think they are needed are they?
Cheers - Bob

**Reply by KarlStraussberger** [reply]
Hi Bob,

Two hints that might help to sort your issues out:
- `<resources>` have to be defined inside `<import>` which has to be in `<config>`. 
- All the artwork is now scanned on import (only once instead of on each browse like before). If you upgrade from before 1.9 you have to rerun the import on your media after having set the `<resources>` correctly.

**Reply by BobManMasher** [reply]
I have the following within <config>tags and the server is starting without error. I force a rescan but the on;y album art I can see when I look at the database is on tracks that have it embereded.  This is true for 1.8 and 1.9 but I can always see the art in the pc Directory. I have also tried with and without container and reseource tags and I am not using subtitles so they haven't been included.

I am also not getting all the Album Artist as you can see from the attached screen shot at the end, 1.9 (left) shows some of the album artists but not all but 1.8 (right) does. Looking at the MP3 Tags - it looks like the two albums highlighted have contributing artist set but not album artist so I need to script this I guess.

    <scripting script-charset="UTF-8">
      <common-script>/usr/local/share/gerbera/js/common.js</common-script>
      <playlist-script>/usr/local/share/gerbera/js/playlists.js</playlist-script>
      <virtual-layout type="builtin">
        <import-script>/usr/local/share/gerbera/js/import.js</import-script>
      </virtual-layout>
    </scripting>

    <resources case-sensitive="no">
        <container>
            <add-file name="cover.jpg"/>
            <add-file name="cover.jpeg"/>
        </container>
        <fanart>
            <container location="image" parentCount="1" minDepth="2"/>
            <add-file name="cover.jpg"/>
            <add-file name="cover.jpeg"/>
        </fanart>
        <resource>
            <add-file name="cover.jpg"/>
            <add-file name="cover.jpeg"/>

        </resource>
    </resources>

![image](https://user-images.githubusercontent.com/77814276/144930071-6a9fb228-f329-4fb2-a095-0ab70460604a.png)

**Reply by KarlStraussberger** [reply]
Seems that your files do not set album artist. In this case there is another new option you can use to pull these properties
https://docs.gerbera.io/en/latest/config-server.html#upnp.

Anyway, I advise to clear your whole database (or use a new one to set everything up)

**Reply by BobManMasher** [reply]
Sorry for the delay getting back - deleting the database and starting again fixed both problems.  That database was created by 1.9.0 which I downloaded but had lots of problems with. So far, my compiled 1.9.2 is working a treat. Thanks for your help.
